docs/ceph-csi-upgrade.md

```diff
@@ -0,0 +1,334 @@
+service/csi-cephfsplugin-provisioner configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+deployment.apps/csi-cephfsplugin-provisioner configured
+```
+
+**Note:** Ignore the warning messages
+
+wait for the deployment to complete
+
+```bash
+$]kubectl get deployment
+NAME                           READY   UP-TO-DATE   AVAILABLE   AGE
+csi-cephfsplugin-provisioner   3/3     1            3           104m
+```
+
+#### 2. CSI upgrade issue
+
+In some scenarios there is an issue in the CSI driver that will cause
+application pods to be disconnected from their mounts when the CSI driver is
+restarted. Since the upgrade would cause the CSI driver to restart if it is
+updated, you need to be aware of whether this affects your applications. This
+issue will happen when using the Ceph fuse client or rbd-nbd:
+
+CephFS: If you are provision volumes for CephFS and have a kernel less than
+version 4.17, The CSI driver will fall back to use the FUSE client.  RBD: If
+you have set the mounter: rbd-nbd option in the RBD storage class, the NBD
+mounter will have this issue.
+
+If you are affected by this issue, you will need to proceed carefully during
+the upgrade to restart your application pods. The recommended step is to modify
+the update strategy of the CSI nodeplugin daemonsets to OnDelete so that you
+can control when the CSI driver pods are restarted on each node.
+
+To avoid this issue in future upgrades, we recommend that you do not use the
+fuse client or rbd-nbd as of now.
+
+#### 3. Upgrade CephFS Nodeplugin resources
+
+upgrading nodeplugin resources include updating the nodeplugin RBAC and
+nodeplugin daemonset
+
+##### 3.1 Update the CephFS Nodeplugin RBAC
+
+```bash
+$]kubectl apply -f deploy/cephfs/kubernetes/v1.14+/csi-nodeplugin-rbac.yaml
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+serviceaccount/cephfs-csi-nodeplugin configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrole.rbac.authorization.k8s.io/cephfs-csi-nodeplugin configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrole.rbac.authorization.k8s.io/cephfs-csi-nodeplugin-rules configured
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+clusterrolebinding.rbac.authorization.k8s.io/cephfs-csi-nodeplugin configured
+```
+
+**Note:** Ignore the warning messages
+
+If you determined in step 2 that you were affected by the CSI driver restart
+issue that disconnects the application pods from their mounts, continue with
+this section. Otherwise, you can skip to step 3.2
+
+```console
+vi deploy/cephfs/kubernetes/v1.14+/csi-cephfsplugin.yaml
+```
+
+```yaml
```

Comment:
It would be nicer to present this as a diff that can be patched to the deployment. Currently the change to the deployment is not clear unless we inspect existing deployment manifest. (same for yaml in line 283)